src/state/Maps/reducers.js

```diff
@@ -241,6 +242,56 @@ const removeMapLayers = (state, mapKey, layerKeys) => {
 	}
 };
 
+/**
+ * Remove layers satisfying filter from map
+ * @param state {Object}
+ * @param mapKey {string}
+ * @param filter {object}
+ * @return {Object} Updated state
+ */
+const removeMapLayersByFilter = (state, mapKey, filter) => {
+	if (!_isEmpty(filter) && mapKey) {
+		const mapLayers = state.maps[mapKey]?.data?.layers;
+		const mapLayersByFilterIndexes = [];
+		mapLayers?.forEach((l, i) => {
+			if (_isMatch(l, filter)) {
+				mapLayersByFilterIndexes.push(i);
+				return true;
+			} else {
+				return false;
+			}
+		});
+
+		if (mapLayersByFilterIndexes?.length) {
+			let updatedLayers = [...state.maps[mapKey]?.data.layers];
+			mapLayersByFilterIndexes.forEach((li, i) => {
```

Comment:
use layer here instead of li for better readability

Reply:
fixed a9042705dd2329ff5eb7667ee0b72c3f45059c8f